File: pset6/sentimental-mario-less/mario.py
# asking user for height
import cs50
while (True):
    height = cs50.get_int("Height: ")
    if height > 0 and height < 9:
        break


def main():
    for i in range(height):
        print_rows(height - i, i + 1)


def print_rows(bricks, space):
    # print spaces
    for m in range(1, bricks, 1):
        print(" ", end="")
    for j in range(space):
        print("#", end="")
    print("")


if __name__ == "__main__":
    main()

File: pset1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_rows(int bricks, int space);

int main(void)
{
    // ask user for the height of the pyramid
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1);

    // print the pyramid
    for (int i = 0; i < n; i++)
    {
        // print pyramid rows
        print_rows(n - i, i + 1);
    }
}

void print_rows(int bricks, int space)
{
    // print spaces
    for (int m = 1; m < bricks; m++)
    {
        printf(" ");
    }

    // print #
    for (int j = 0; j < space; j++)
    {
        printf("#");
    }
    printf("\n");
}
